native-media/app/src/main/java/com/example/nativemedia/NativeMedia.java

```diff
@@ -206,10 +214,19 @@ public void onNothingSelected(AdapterView parent) {
         // initialize button click handlers
 
         // Java MediaPlayer start/pause
-
+        final Activity appActivity = this;
         ((Button) findViewById(R.id.start_java)).setOnClickListener(new View.OnClickListener() {
 
             public void onClick(View view) {
+                // Request the READ_EXTERNAL_STORAGE permission
+                if (ActivityCompat.checkSelfPermission(appActivity, Manifest.permission.READ_EXTERNAL_STORAGE)
```

Comment:
You could move this check and request into a dedicated method (returning a boolean to indicate whether the permission is missing and has been requested.)
It's the same check as in the other clickListener.